server/src/main/java/org/opensearch/action/search/CreatePITResponse.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.opensearch.action.ActionResponse;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.StatusToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.rest.action.RestActions;
+
+import java.io.IOException;
+
+/**
+ * Create point in time response with point in time id and shard success / failures
+ */
+public class CreatePITResponse extends ActionResponse implements StatusToXContentObject {
+    // point in time id
+    private final String id;
+    private final int totalShards;
+    private final int successfulShards;
+    private final int failedShards;
+    private final int skippedShards;
+    private final ShardSearchFailure[] shardFailures;
+
+    public CreatePITResponse(SearchResponse searchResponse) {
```

Comment:
this sets a risky precedent. Let's atleast assert that searchResponse.pointInTimeId() is non-empty.